runtime/dolphin/src/lib.rs

```diff
@@ -1107,9 +1139,14 @@ impl FungibleLedger<Runtime> for MantaFungibleLedger {
 		beneficiary: &<Runtime as frame_system::Config>::AccountId,
 		amount: Balance,
 	) -> Result<(), FungibleLedgerConsequence> {
+		Self.is_valid(asset_id)?;
 		Self::can_deposit(asset_id, beneficiary, amount)?;
-		if asset_id == 0 {
-			let _ = <Balances as Currency<<Runtime as frame_system::Config>::AccountId>>::deposit_creating(beneficiary, amount);
+		if asset_id == <MantaAssetConfig as AssetConfig<Runtime>>::NativeAssetId::get() {
+			let _ =
```

Comment:
Why ignore the error?

Reply:
The returned type is not really an error, instead it is an over-engineered `Imbalance` type (at least in my view). I will put a FIXME here.